src/cat-data-service/src/service/v0/snapshot.rs

```diff
@@ -1,52 +1,62 @@
-use crate::db::snapshot::SnapshotDb;
+use crate::{
+    service::{handle_result, Error},
+    state::State,
+};
 use axum::{extract::Path, routing::get, Router};
 use std::sync::Arc;
 
-pub fn snapshot<State: SnapshotDb + Send + Sync + 'static>(state: Arc<State>) -> Router {
+pub fn snapshot(state: Arc<State>) -> Router {
     Router::new()
         .route(
             "/snapshot",
             get({
                 let state = state.clone();
-                move || versions_exec(state)
+                move || async { handle_result(versions_exec(state).await).await }
             }),
         )
         .route(
             "/snapshot/voter/:event/:voting_key",
             get({
                 let state = state.clone();
-                move |path| voter_exec(path, state)
+                move |path| async { handle_result(voter_exec(path, state).await).await }
             }),
         )
         .route(
             "/snapshot/delegator/:event/:stake_public_key",
-            get(move |path| delegator_exec(path, state)),
+            get(move |path| async { handle_result(delegator_exec(path, state).await).await }),
         )
 }
 
-async fn versions_exec<State: SnapshotDb>(state: Arc<State>) -> String {
+async fn versions_exec(state: Arc<State>) -> Result<String, Error> {
     tracing::debug!("versions_exec");
 
-    let snapshot_versions = state.get_snapshot_versions();
-    serde_json::to_string(&snapshot_versions).unwrap()
+    let snapshot_versions = state.event_db.get_snapshot_versions().await?;
+    Ok(serde_json::to_string(&snapshot_versions).unwrap())
 }
 
-async fn voter_exec<State: SnapshotDb>(
+async fn voter_exec(
     Path((event, voting_key)): Path<(String, String)>,
     state: Arc<State>,
-) -> String {
+) -> Result<String, Error> {
     tracing::debug!("voter_exec, event: {0}, voting_key: {1}", event, voting_key);
 
-    let voter = state.get_voter(event, voting_key);
-    serde_json::to_string(&voter).unwrap()
+    let voter = state.event_db.get_voter(event, voting_key).await?;
+    Ok(serde_json::to_string(&voter).unwrap())
 }
 
-async fn delegator_exec<State: SnapshotDb>(
+async fn delegator_exec(
     Path((event, stake_public_key)): Path<(String, String)>,
     state: Arc<State>,
-) -> String {
-    tracing::debug!("delegator_exec, event: {0}, stake_public_key: {1}", event, stake_public_key);
+) -> Result<String, Error> {
+    tracing::debug!(
+        "delegator_exec, event: {0}, stake_public_key: {1}",
+        event,
+        stake_public_key
+    );
 
-    let delegator = state.get_delegator(event, stake_public_key);
-    serde_json::to_string(&delegator).unwrap()
+    let delegator = state
+        .event_db
+        .get_delegator(event, stake_public_key)
+        .await?;
+    Ok(serde_json::to_string(&delegator).unwrap())
```

Comment:
We don't need to do the string conversion here - we can just return `Json(delegator)` (`Json` comes from axum)